akka-slf4j/src/main/scala/akka/event/slf4j/Slf4jLogger.scala

```diff
@@ -104,18 +104,15 @@ class Slf4jLogger extends Actor with SLF4JLogging with RequiresMessageQueue[Logg
 
   @inline
   final def withMdc(logSource: String, logEvent: LogEvent)(logStatement: => Unit): Unit = {
-    MDC.put(mdcAkkaSourceAttributeName, logSource)
-    MDC.put(mdcThreadAttributeName, logEvent.thread.getName)
-    MDC.put(mdcAkkaTimestamp, formatTimestamp(logEvent.timestamp))
-    MDC.put(mdcActorSystemAttributeName, actorSystemName)
-    logEvent.mdc.foreach { case (k, v) => MDC.put(k, String.valueOf(v)) }
+    val mdcAdapter = MDC.getMDCAdapter
+    mdcAdapter.put(mdcAkkaSourceAttributeName, logSource)
+    mdcAdapter.put(mdcThreadAttributeName, logEvent.thread.getName)
+    mdcAdapter.put(mdcAkkaTimestamp, formatTimestamp(logEvent.timestamp))
+    mdcAdapter.put(mdcActorSystemAttributeName, actorSystemName)
+    logEvent.mdc.foreach { case (k, v) => mdcAdapter.put(k, String.valueOf(v)) }
     try logStatement
     finally {
-      MDC.remove(mdcAkkaSourceAttributeName)
-      MDC.remove(mdcThreadAttributeName)
-      MDC.remove(mdcAkkaTimestamp)
-      MDC.remove(mdcActorSystemAttributeName)
-      logEvent.mdc.keys.foreach(k => MDC.remove(k))
+      MDC.clear()
```

Comment:
If there were some other MDC values in the ThreadLocal that would be invalid usage of MDC so should be fine to clear the whole thing. Also, I think removing all entries still leaves something in the Thread so a small leak to not clear.

Reply:
@johanandren what do you think I should do with this PR? I still think the MDC.clear() at the end is more correct and more efficient. Using the mdcAdapter for the put doesn't help much so I can revert that part?

Reply:
Sounds good to me.

Reply:
done, also changed other places that used mdcAdapter.put

Reply:
:+1: